shipment.go

```diff
@@ -400,3 +424,31 @@ func (c *Client) GetShipmentEstimatedDeliveryDateWithContext(ctx context.Context
 	err = c.do(ctx, http.MethodGet, "shipments/"+shipmentID+"/smartrate/delivery_date", vals, &res)
 	return
 }
+
+// EstimateDeliveryDateForShipment retrieves the estimated delivery date of each rate for a Shipment via the Delivery Date Estimator API, based on a specific ship date.
+func (c *Client) EstimateDeliveryDateForShipment(shipmentID string, params *EstimateDeliveryDateForShipmentParams) (out []*EstimateDeliveryDateForShipmentResult, err error) {
+	return c.EstimateDeliveryDateForShipmentWithContext(context.Background(), shipmentID, params)
+}
+
+// EstimateDeliveryDateForShipmentWithContext performs the same operation as EstimateDeliveryDateForShipment, but allows specifying a context that can interrupt the request.
+func (c *Client) EstimateDeliveryDateForShipmentWithContext(ctx context.Context, shipmentID string, params *EstimateDeliveryDateForShipmentParams) (out []*EstimateDeliveryDateForShipmentResult, err error) {
+	res := struct {
+		Results *[]*EstimateDeliveryDateForShipmentResult `json:"rates,omitempty"`
+	}{Results: &out}
+	err = c.do(ctx, http.MethodGet, "shipments/"+shipmentID+"/smartrate/delivery_date", params, &res)
+	return
+}
+
+// RecommendShipDateForShipment retrieves the recommended ship date of each rate for a Shipment via the Precision Shipping API, based on a specific desired delivery date.
+func (c *Client) RecommendShipDateForShipment(shipmentID string, params *RecommendShipDateForShipmentParams) (out []*RecommendShipDateForShipmentResult, err error) {
```

Comment:
```suggestion
func (c *Client) RecommendShipDateForShipment(shipmentID string, desiredDeliveryDate string) (out []*RecommendShipDateForShipmentResult, err error) {
```

This should just be a string instead of params